src/dataplane-protocol/src/batch.rs

```diff
@@ -15,10 +15,7 @@ use crate::derive::Encode;
 
 use crate::Offset;
 use crate::Size;
-use crate::record::DefaultRecord;
-
-pub type DefaultBatchRecords = Vec<DefaultRecord>;
-pub type DefaultBatch = Batch<DefaultBatchRecords>;
```

Comment:
We still need to keep this for compatibility, can mark as deprecated

Reply:
Ok, but just out of curiosity, what sort of compatibility are you talking about? At the type-level, `DefaultBatch` and `Batch<R = Vec<Record>>` are equivalent, one can be substituted for another. Plus, all of the instances of `DefaultBatch` have been removed. So, I believe we have the following:

- This PR makes breaking changes and we are bumping up the crate version to reflect that
- We are also updating the dependencies to use the new version and fixing breakages
- And, crates using old versions of this crate should still be able to interop just fine, since the type resolution should be equivalent.

So I _think_ that removing these shouldn't cause any harm, but if you want I can keep these around for awhile as deprecated until we're sure.